core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -729,6 +784,8 @@ public boolean isFollowing() {
             Protos.Key.Builder mnemonicEntry = BasicKeyChain.serializeEncryptableItem(seed);
             mnemonicEntry.setType(Protos.Key.Type.DETERMINISTIC_MNEMONIC);
             serializeSeedEncryptableItem(seed, mnemonicEntry);
+            // TODO this is a current workaround!!! Please move it to a dedicated entry
+            mnemonicEntry.setLabel(HDUtils.formatPath(getAccountPath()));
```

Comment:
I originally wanted to change as little as possible and keep the code as it was done in previous commits.
Happy to make changes towards a more sensible architecture for this.

Reply:
Thanks, it's much appreciated. Before you continue: What do you think? Can we do without persisting the path for mnemonic keys, because we can derive it on the fly? Or is it mandatory to store it?

Reply:
I actually think it's already stored in that part of the code (last lines)
```java
protected List<Protos.Key> serializeMyselfToProtobuf() {
        // Most of the serialization work is delegated to the basic key chain, which will serialize the bulk of the
        // data (handling encryption along the way), and letting us patch it up with the extra data we care about.
        LinkedList<Protos.Key> entries = newLinkedList();
        if (seed != null) {
            Protos.Key.Builder mnemonicEntry = BasicKeyChain.serializeEncryptableItem(seed);
            mnemonicEntry.setType(Protos.Key.Type.DETERMINISTIC_MNEMONIC);
            serializeSeedEncryptableItem(seed, mnemonicEntry);
            entries.add(mnemonicEntry.build());
        }
        Map<ECKey, Protos.Key.Builder> keys = basicKeyChain.serializeToEditableProtobufs();
        for (Map.Entry<ECKey, Protos.Key.Builder> entry : keys.entrySet()) {
            DeterministicKey key = (DeterministicKey) entry.getKey();
            Protos.Key.Builder proto = entry.getValue();
            proto.setType(Protos.Key.Type.DETERMINISTIC_KEY);
            final Protos.DeterministicKey.Builder detKey = proto.getDeterministicKeyBuilder();
            detKey.setChainCode(ByteString.copyFrom(key.getChainCode()));
            for (ChildNumber num : key.getPath())
                detKey.addPath(num.i());
```

But it seems it's actually never read which is quite puzzling to me. Any idea why ?